### ISSUE TITLE
No more ZTS support

### ISSUE THREAD

**Issue by Hast** [oerdnj/deb.sury.org#327]
Is there any plans to keep PHP 7 Thread Safe version up to date? Why have you stopped supporting it?

**Comment by oerdnj** [comment]
@Hast The build is broken and I haven't had time to fix it yet.

**Comment by oerdnj** [comment]
@Hast Try now - only trusty and xenial and only amd64 is provided now. No external PECL modules (but you can always compiled those yourself).

**Comment by Hast** [comment]
@oerdnj it seems to be working on 14.4, thanks.

**Comment by JesusTheHun** [comment]
@oerdnj No phpize7.0-zts is available, I've made one from phpize7.0 but /usr/lib/php/20151012-zts/**build** is missing so we can't compile PEC modules

**Comment by oerdnj** [comment]
I probably won't look into that very soon, so patches to fix this would be welcomed.

**Comment by JesusTheHun** [comment]
I can't fix the package right now because a bug that seems related to debhelper. 
Also If you had a git repo of your package source that would help a lot.

**Comment by oerdnj** [comment]
The master repo: https://anonscm.debian.org/git/pkg-php/php.git and github mirror: https://github.com/pkg-php/php

PHP 7.0 is in the `master-7.0` branch...

**Comment by JesusTheHun** [comment]
Ok thx, what is your debhelper version ?
debian/7.0.6-9 fails to compile with : 

```
configure: exit 1
	cd /home/jmassuch/Downloads/php-debian-7.0.6-9
Died at /usr/share/perl5/Debian/Debhelper/Buildsystem/autoconf.pm line 77.
debian/rules:320 : la recette pour la cible « override_dh_auto_configure-arch » a échouée
make[1]: *** [override_dh_auto_configure-arch] Erreur 255
```

/usr/share/perl5/Debian/Debhelper/Buildsystem/autoconf.pm belongs to debhelper package.

```
$ apt-cache policy debhelper
debhelper:
  Installé : 9.20160115ubuntu3
  Candidat : 9.20160115ubuntu3
 Table de version :
 *** 9.20160115ubuntu3 500
        500 http://distrib-coffee.ipsl.jussieu.fr/pub/linux/ubuntu xenial/main amd64 Packages
        500 http://distrib-coffee.ipsl.jussieu.fr/pub/linux/ubuntu xenial/main i386 Packages
        100 /var/lib/dpkg/status
```

**Comment by oerdnj** [comment]
Could you run:
```
apt-get install git-buildpackage
gbp clone https://anonscm.debian.org/git/pkg-php/php.git 
cd php
LC_ALL=C.UTF-8 gbp buildpackage --export-dir=../build-area
```

And paste full `gbp buildpackage` build log (found in `../build-area/*.build`) as a gist?

**Comment by JesusTheHun** [comment]
Due to company policy I gave to clone repo manually, using :

`$ LC_ALL=C.UTF-8 gbp buildpackage --export-dir=../build-area`

I have : 

```
gbp:warning: Pristine-tar branch "pristine-tar" not found
gbp:error: Pristine-tar couldn't checkout "php7.0_7.0.6.orig.tar.gz": fatal: Path 'php7.0_7.0.6.orig.tar.gz.delta' does not exist in 'refs/remotes/origin/pristine-tar'
pristine-tar: git show refs/remotes/origin/pristine-tar:php7.0_7.0.6.orig.tar.gz.delta failed
```

**Comment by oerdnj** [comment]
Ok, you need to also checkout these branches:
* `pristine-tar`
* `upstream-7.0`

and then switch back to `master-7.0`

**Comment by JesusTheHun** [comment]
```
$ LC_ALL=C.UTF-8 gbp buildpackage --export-dir=../build-area
debuild: unknown dpkg-buildpackage/debuild option: --export-dir=../build-area
 dpkg-buildpackage -rfakeroot -D -us -uc -i -I --export-dir=../build-area
dpkg-buildpackage: unknown option or argument --export-dir=../build-area

Use --help for program usage information.
debuild: fatal error at line 1376:
dpkg-buildpackage -rfakeroot -D -us -uc -i -I --export-dir=../build-area failed
```

I can't find anything similar to ---export-dir in `man dpkg-buildpackage`

**Comment by oerdnj** [comment]
Sorry, it should be `--git-export-dir`

**Comment by JesusTheHun** [comment]
Compiling so far... It's in a VM so it will take some times. I should be able to manage a fix for a the ZTS version to allow PECL compilations. Eventually with some hints from you I could also allow some PECL packages couldn't I ?

**Comment by oerdnj** [comment]
Yup, look here: https://anonscm.debian.org/git/pkg-php/php-defaults.git and compare `master` and `master-coinst` branches. You'll need to modify `debian/rules`:
```
PHP_DEFAULT_VERSION    := 7.0
PHP_SUPPORTED_VERSIONS := 7.0
```
to `7.0-zts` and then run:

`debian/rules debian/control`

and then build the `src:php-defaults` packages...

**Comment by oerdnj** [comment]
I think it should be very easy to fix, something somewhere is missing `$(ZTS)` either in `debian/rules` or `debian/*.install`.

**Comment by oerdnj** [comment]
And here's my post on packaging new extensions: https://lists.alioth.debian.org/pipermail/pkg-php-pecl/Week-of-Mon-20160425/001182.html

**Comment by JesusTheHun** [comment]
You lost me.
There are two repository ?
What are their uses/diffrences ?
On what repository shall I make changes ?

There is no `PHP_DEFAULT_VERSION := 7.0` in https://github.com/pkg-php/php/blob/master-7.0/debian/rules nor in `/usr/share/dpkg/default.mk`

You made master-7.0 so you can compile ZTS directly from it (afaik reading debian/rules), how can I trigger ZTS compilation directly from this main branch ? Is it really wise to do such ?

PS : non-zts compilation is fine

**Comment by compunect** [comment]
With the new "parallel" for PHP 8 threads are on the way to become mainstream.
So much work went into making threads easy to implement but without ZTS support it's all for nothing

**Comment by williamdes** [comment]
> With the new "parallel" for PHP 8 threads are on the way to become mainstream. So much work went into making threads easy to implement but without ZTS support it's all for nothing

Just ran in this now
@oerdnj if contributions are welcome to bring back zts builds should I proceed ?

**Comment by lolcatw** [comment]
Any updates on this?

**Comment by aviramaz** [comment]
+1

**Comment by realjjaveweb** [comment]
# BOUNTY
I **will send ~$20/€20 bucks** to anyone who will create an |**acceptable** && **_afterwards merged && working| (ALL conditions must apply!) PR_**  that will end up with a PHP8.1(+) ZTS (Zend Thread Safe) enabled package in the `deb.sury.org` repository. If there will be even package that would include or would install automatically `\parallel` ([this parallel](https://www.php.net/manual/en/book.parallel.php)) into it, I will throw in another ~$10/€10 bucks. (same conditions)
